include/net/buf.h

```diff
@@ -590,7 +601,9 @@ struct net_buf *net_buf_get(struct k_fifo *fifo, s32_t timeout);
  */
 static inline void net_buf_destroy(struct net_buf *buf)
 {
-	k_lifo_put(&buf->pool->free, buf);
+	struct net_buf_pool *pool = net_buf_pool_get(buf->pool_id);
```

Comment:
If we are not checking the return value of net_buf_pool_get(), then why not write

k_lifo_put(&net_buf_pool_get(buf->pool_id)->free, buf)

instead and save extra variable here.